src/handlers/connectionHandler.ts

```
@@ -28,115 +49,212 @@ function setConnectionEstablishmentTimeout(timeoutMs: number = RECONNECT_CONFIG.
+	botLogger.info(`ℹ️ Connection status change: ${ctx.status}`, {
 		status: ctx.status,
 		currentRetries: RECONNECT_CONFIG.currentRetries
 	});
 
 	switch (ctx.status) {
 		case 'connecting':
-			botLogger.connection("Connecting to WhatsApp...");
-			// Set timeout to monitor this connection attempt
-			setConnectionEstablishmentTimeout();
+			const now = Date.now();
+			const timeSinceLastConnectingEvent = now - lastConnectingEvent;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)